replica/table.cc

```diff
@@ -856,18 +856,17 @@ table::sstable_list_builder::build_new_list(const sstables::sstable_set& current
 }
 
 future<>
-table::update_sstable_lists_on_off_strategy_completion(const std::vector<sstables::shared_sstable>& old_maintenance_sstables,
-                                                       const std::vector<sstables::shared_sstable>& new_main_sstables) {
+table::update_sstable_lists_on_off_strategy_completion(sstables::compaction_completion_desc desc) {
     class sstable_lists_updater : public row_cache::external_updater_impl {
         using sstables_t = std::vector<sstables::shared_sstable>;
         table& _t;
         sstable_list_builder _builder;
-        const sstables_t& _old_maintenance;
-        const sstables_t& _new_main;
+        const sstables_t _old_maintenance;
+        const sstables_t _new_main;
         lw_shared_ptr<sstables::sstable_set> _new_maintenance_list;
         lw_shared_ptr<sstables::sstable_set> _new_main_list;
     public:
-        explicit sstable_lists_updater(table& t, sstable_list_builder::permit_t permit, const sstables_t& old_maintenance, const sstables_t& new_main)
+        explicit sstable_lists_updater(table& t, sstable_list_builder::permit_t permit, const sstables_t old_maintenance, const sstables_t new_main)
                 : _t(t), _builder(std::move(permit)), _old_maintenance(old_maintenance), _new_main(new_main) {
```

Comment:
If we copy them, why pass by value?
If passing by value, why not move the args?

Reply:
you're right. will move.